From cfba1434e3aea571b9be9e2315a8df66d0c8f5a4 Mon Sep 17 00:00:00 2001
From: Felix Wruck <felix.wruck@aisec.fraunhofer.de>
Date: Fri, 19 Sep 2025 10:49:49 +0200
Subject: [PATCH] flash-var, sign-efi-sig-list: Fix include for strptime
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Currently, builds with strict GCC flags fail due to an
"error: implicit declaration of function ‘strptime’".

Therefore, add the missing #define _XOPEN_SOURCE in flash-var.c
and sign-efi-sig-list.c before including time.h.

Signed-off-by: Felix Wruck <felix.wruck@aisec.fraunhofer.de>
---
 flash-var.c         | 1 +
 sign-efi-sig-list.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/flash-var.c b/flash-var.c
index aa10ae6..d5f5f2f 100644
--- a/flash-var.c
+++ b/flash-var.c
@@ -1,3 +1,4 @@
+#define _XOPEN_SOURCE
 #include <stdlib.h>
 #include <stdint.h>
 #include <sys/types.h>
diff --git a/sign-efi-sig-list.c b/sign-efi-sig-list.c
index 94bd7d4..c096c70 100644
--- a/sign-efi-sig-list.c
+++ b/sign-efi-sig-list.c
@@ -3,6 +3,7 @@
  *
  * see COPYING file
  */
+#define _XOPEN_SOURCE
 #include <stdint.h>
 #define __STDC_VERSION__ 199901L
 #include <efi.h>
-- 
2.47.3

